### PR TITLE
chore: automate server release on hosted with prerelease

### PR DESCRIPTION
**Pull request by mihir20**
# Description

automate server release on hosted whenever we push to a pre-release branch

## Linear Ticket

https://linear.app/rudderstack/issue/PIPE-1327/trigger-release-workflow-from-pre-release

## Security

- [x] The code changed/added as part of this pull request won't create any security issues with how the software is being used.

### CODE REVIEW

File: .github/workflows/prerelease.yaml [hunk 1]
```diff
@@ -52,3 +53,19 @@ jobs:
...
           prerelease: true
           release-as: ${{ steps.extract_branch.outputs.branch }}
+      - name: Extract prerelease version
+        shell: bash
+        run: echo "version=$(echo "${{ steps.extract_branch.outputs.branch }}" | cut -d'/' -f2)" >> $GITHUB_OUTPUT
+        id: extract_version
+      - name: Trigger dispatch event
+        uses: peter-evans/repository-dispatch@v3
+        # release please run 2 times, first for creating the PR and second for creating the release
+        if: ${{ steps.release_action.outputs.releases_created == 'true' }}
```
Comment (ktgowtham) [inline .github/workflows/prerelease.yaml:63]: Didn't really understand. the event trigger that's configured is a commit on prerelease/* branch. why will this run for release?

Comment (mihir20) [reply .github/workflows/prerelease.yaml:63]: when we create the new branch this action will run and create a [pull req](https://github.com/rudderlabs/rudder-server/pull/4911) with all the change logs:
https://github.com/rudderlabs/rudder-server/actions/runs/10037166229
Once we merge that [PR](https://github.com/rudderlabs/rudder-server/pull/4911) this action will run again and create a release
https://github.com/rudderlabs/rudder-server/actions/runs/10037739692

Comment (ktgowtham) [reply .github/workflows/prerelease.yaml:63]: make sense. I missed the new PR 🤦 ... I was thinking of only new commits that are added to prerelease/* branch

**Review by ktgowtham — APPROVED** [review]
LGTM

**Review by lvrach — APPROVED** [review]

**Pull request merged**